server/lib/nl/common/utils.py

```diff
@@ -121,19 +166,13 @@ def sv_existence_for_places_check_single_point(
 
   existing_svs = {}
   existsv2places = {}
+  facet_info = series_facet.get('facets', {})
   for sv, sv_data in series_facet.get('data', {}).items():
     for pl, place_data in sv_data.items():
       for facet_data in place_data:
-        # If there is a date specified, but facet dates are different
-        # granularity, move on to next facet
-        # TODO: allow for different granularity but same year.
-        if date and (len(date) != len(facet_data.get('earliestDate', '')) or
-                     len(date) != len(facet_data.get('latestDate', ''))):
-          continue
-        # If there is a date specified, but it is outside the range of dates of
-        # this facet, move on to next facet
-        if date and (date < facet_data.get('earliestDate', '') or
-                     date > facet_data.get('latestDate', '')):
+        # Check that this facet has data for specified date (if no specified date, facet_contains_data will return true)
+        if not _facet_contains_date(
+            facet_data, facet_info.get(facet_data.get('facet', ''), {}), date):
```

Comment:
nit: use a local var for `facet_info.get(facet_data.get('facet', ''), {})`?

Reply:
done